.github/workflows/publish.yml

```diff
@@ -0,0 +1,107 @@
+---
+name: Publish on Test PyPI and PyPI
+
+on:
+    push:
+        branches:
+      # Commits pushed to release/ branches are published on Test PyPI if they
+      # have a new version number.
+            - release/**
+        tags:
+      # Tags that start with the "v" prefix are published on PyPI.
+            - v*
+
+jobs:
+
+    build:
+
+        runs-on: ubuntu-latest
+
+        steps:
+
+            - uses: actions/checkout@v2
+
+            - name: Set up Python 3.10
+              uses: actions/setup-python@v2
+              with:
+                  python-version: '3.10'
+
+            - name: Install pypa/build
+              run: python -m pip install build
+
+            - name: Build a binary wheel and a source tarball
+              run: >-
+                  python -m
+                  build
+                  --sdist
+                  --wheel
+                  --outdir dist/
+            - name: Upload distribution artifact
+              uses: actions/upload-artifact@v2
+              with:
+                  name: release
+                  path: dist/
+
+
+    publish-test:
+
+        name: Build and publish on TestPyPI
+        if: >
+            startsWith(github.ref, 'refs/heads/release/') ||
+            startsWith(github.ref, 'refs/tags/')
+        needs: [build]
+        runs-on: ubuntu-latest
+
+        environment:
+            name: Test PyPI
+            url: https://test.pypi.org/project/aiida-cp2k/
+
+        steps:
+            - uses: actions/download-artifact@v2
+              name: Download distribution artifact
+              with:
+                  name: release
+                  path: dist/
+
+            - name: Publish distribution on Test PyPI
+              uses: pypa/gh-action-pypi-publish@release/v1
+              if: startsWith(github.ref, 'refs/heads/release/')
+              with:
+                  user: __token__
+                  password: ${{ secrets.PYPI_API_TOKEN }}
+                  repository_url: https://test.pypi.org/legacy/
+                  skip_existing: true
+
+    publish:
+
+        name: Build and publish on PyPI
+        if: startsWith(github.ref, 'refs/tags')
+
+        needs: [build]
```

Comment:
If this only needs the `build` job, then why still do the `publish-test` job? Won't there be run simultaneously?

Reply:
The `publish-test` will publish things on TestPyPI. It is needed to check that everything is alright, before actual publishing. At the same time the line `startsWith(github.ref, 'refs/tags/')` is probably redundant for the TestPyPI. I will remove it.

Reply:
done.